src/main/java/com/softserve/sprintone/ivan/TaskIvanSprintOne.java

```diff
@@ -0,0 +1,422 @@
+
+/**
+ * @author Iván Caballero
+ */
+
+/**
+ * Task 13 - The code provided has a method plus100 which is supposed to return the number of the
+ * parameter incremented by 100. But it's not working properly.
+ * Constraints:
+ * - Do not modify the constructor
+ * - Do not modify the plus100 method
+ * - Do not modify the ONE_HUNDRED declaration
+ */
+class Dinglemouse {
+    private static final int ONE_HUNDRED = 100;
+    public static final Dinglemouse dinglemouse = new Dinglemouse();
+    private final int value;
+
+    private Dinglemouse() {
+        value = ONE_HUNDRED;
+    }
+
+    public int plus100(int n) {
+        return value + n;
+    }
+}
+
+/**
+ * Class TaskIvanSprintOne set all the tasks to accomplish
+ */
+public class TaskIvanSprintOne {
+    /**
+     * Class Encoder needed for task 49
+     * Specific instructions:
+     * - The code must contain a Leetspeak class inherited from an Encoder abstract class and containing an
+     * encode(String) method returning itself the encoded String.
+     * - The encode method should return an empty String if a null parameter is passed.
+     * - If any uppercase character is given, it should be compared also with the dictionary,
+     * 'a' gives '4' but 'A' gives also '4'.
+     * - If any character outside the dictionary is given, it should be output as is.
+     */
+    abstract static class Encoder{
+        public abstract String encode(String source);
+    }
+
+    public static class Leetspeak extends Encoder{
+        private List<String> characters;
+
+        private List<String> encodeCharacters;
+
+        public Leetspeak(){
+            characters = new ArrayList<>(){{
+                            add("a");
+                            add("e");
+                            add("l");
+                            add("m");
+                            add("o");
+                            add("u");
+                        }};
+            encodeCharacters = new ArrayList<>(){{
+                                    add("4");
+                                    add("3");
+                                    add("1");
+                                    add("/^^\\");
+                                    add("0");
+                                    add("(_)");
+                                }};
+        }
+
+        public String encode(String source){
+            StringBuilder st = new StringBuilder();
+            int index;
+
+            if (source.equals(""))
+                return null;
+
+            for (String s : source.split("")) {
+                if (characters.contains(s.toLowerCase())){
+                    index = characters.indexOf(s.toLowerCase());
+                    st.append(encodeCharacters.get(index));
+                } else {
+                    st.append(s);
+                }
+            }
+
+            return st.toString();
+        }
+
+    }
+
+    public static final TaskIvanSprintOne INST = new TaskIvanSprintOne();
```

Comment:
did not get for what you would be using an instance of the same class? I don't see recurrency is needed.